client/network/src/block_request_handler.rs

```diff
@@ -228,21 +215,6 @@ impl<B: BlockT> BlockRequestHandler<B> {
 				support_multiple_justifications,
 			)?;
 
-			// If any of the blocks contains nay data, we can consider it as successful request.
-			if block_response
-				.blocks
-				.iter()
-				.any(|b| !b.header.is_empty() || !b.body.is_empty() || b.is_empty_justification)
-			{
-				if let Some(value) = self.seen_requests.get_mut(&key) {
-					// If this is the first time we have processed this request, we need to change
-					// it to `Fulfilled`.
-					if let SeenRequestsValue::First = value {
-						*value = SeenRequestsValue::Fulfilled(1);
-					}
```

Comment:
Here we changed from `First` to `Fulfilled(1)`.

Reply:
That's right, but only on `If any of the blocks contains nay data, we can consider it as successful request.`. In case there is no data, you can spam a node as much as you want.

Reply:
I know. But answering a request thet resolves to nothing should be cheap.

Reply:
However in case of `Some(SeenRequestsValue::Fulfilled(ref mut requests))` it is incremented on every request.

Reply:
The first time we have fulfilled the request, means that we also fulfill every request after it

Reply:
There is for example the justification request that is send sometimes before the other node already has the justification, but after the first time we have send back the justification we should not request it anymore.